### DISCUSSION TITLE
SQL syntax highlight in strings

### DISCUSSION THREAD

**Discussion by alippai** [microsoft/pylance-release#2591] [Enhancement]
https://github.com/microsoft/vscode/issues/1751 this is the only related issue I found, but it's scope is huge.

I was thinking whether a simple (eg. regex based) syntax highlight for strings and f-strings containing SQL would be possible in pylance (instead of the universal injectibility of any languange). It works well in eg. PyCharm.

I don't expect SQL syntax check or SQL format options which a SQL language flavor aware plugin would be able to do, just some basic detection of literals, functions and ANSI keywords.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by luabud** [comment]
I believe there's an extension that already does this: https://marketplace.visualstudio.com/items?itemName=qufiwefefwoyn.inline-sql-syntax
